apps/wallet/src/ui/app/shared/InputWithAction.tsx

```diff
@@ -15,42 +16,60 @@ export interface InputWithActionProps
     onActionClicked?: PillProps['onClick'];
     actionType?: PillProps['type'];
     name: string;
+    prefix?: string;
+    suffix?: string;
+    actionDisabled?: boolean;
+    allowNegative?: boolean;
 }
 
 export function InputWithAction({
     actionText,
     onActionClicked,
+    allowNegative,
     actionType = 'submit',
     type = 'number',
     disabled = false,
+    actionDisabled = false,
     name,
+    prefix,
+    suffix,
     ...props
 }: InputWithActionProps) {
     const [field, meta] = useField(name);
     const { isSubmitting } = useFormikContext();
     const isInputDisabled = isSubmitting || disabled;
-    const isActionDisabled =
```

Comment:
We could add `actionDisabled` prop to be `auto` or boolean and in case auto keep this logic otherwise whatever the value of `actionDisabled` is. What do you think?